lib/loader/ast/component/component_type.cpp

```diff
@@ -45,7 +46,7 @@ Expect<void> Loader::loadType(ValueType &Ty) {
     Ty.emplace<PrimValType>(static_cast<PrimValType>(Tag));
     break;
   default:
-    Ty.emplace<TypeIndex>(Tag);
+    Ty.emplace<uint32_t>(static_cast<uint32_t>(Tag));
```

Comment:
I recommend left the alias type `TypeIndex` to avoid confusion in the future.

Reply:
I'll refactor it in the future.
Actually, the definition is the same as every `index` in core WASM, and we use `uint32_t` to implement it. That's why I modified it back into `uint32_t`.